httpclient/api_client.go

```diff
@@ -392,7 +392,11 @@ func (c *ApiClient) perform(
 	return resp, nil
 }
 
-func orDefault(configured, _default int) int {
+type number interface {
+	~int | ~int64 | ~float64 | ~float32
+}
+
+func orDefault[T number](configured, _default T) T {
```

Comment:
There are some other users of this function in the package. I opted for generic version so not to change other consumers.